src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -13,6 +13,7 @@
 #include "cluster/partition_manager.h"
 #include "cluster/shard_table.h"
 #include "kafka/protocol/errors.h"
+#include "kafka/server/materialized_partition.h"
```

Comment:
please write in the commit message how and why this changes the behavior of list offsets relative to its previous behavior.